lib/synapse-core/desktop-file-service.vala

```diff
@@ -185,7 +187,18 @@ namespace Synapse {
                     is_hidden = false;
                 }
             } catch (Error err) {
-                critical (err.message);
+                string name = "Unidentified";
+
+                if (keyfile.has_key (GROUP, "Name")) {
```

Comment:
I'm getting compiler warnings about `unhandled error 'GLib.KeyFileError'`

Reply:
Sorry, missed that.  Not immediately obvious why since the original error handling was minimal!

Reply:
Oh - I see. Doh.